R/list.data.R

```diff
@@ -0,0 +1,154 @@
+#' Listing the data for the current project
+#'
+#' This function produces a data.frame of all data files in the project, with
+#'   meta data on if and how the file will be loaded by \code{load.project}.
+#'
+#' @param override.config Named list, allows overriding individual configuration
+#'   items.
+#'
+#' @return A data.frame listing the available data, with relevant meta data
+#'
+#' @details The returned data.frame contains the following variables, with one
+#'   observation per file in \code{data/}:
+#'
+#' \tabular{ll}{
+#'    \code{filename} \tab Character variable containing the filename relative
+#'      to \code{data/} directory. \cr
+#'    \code{varname} \tab Character variable containing the name of the
+#'      variable into which the file will be imported. * \cr
+#'    \code{is_ignored} \tab Logical variable that indicates whether the file.
+#'      is ignored through the \code{data_ignore} option in the configuration \cr
+#'    \code{is_directory} \tab Logical variable that indicates whether the file
+#'      is a directory. \cr
+#'    \code{is_cached} \tab Logical variable that indicates whether the file is
+#'      already available in the \code{cache/} directory. \cr
+#'    \code{cached_only} \tab Logical variable that indicates whether the
+#'      variable is only available in the \code{cache/} directory. This occurs
+#'      when calling the cache function with a code fragment in a munge script. \cr
+#'    \code{reader} \tab Character variable containing the name of the reader
+#'      function that will be used to load the data. Contains a \code{character(0)}
+#'      if no suitable reader was found.
+#' }
+#'
+#' * Note that some readers return more than one variable, usually with the
+#'   listed variablename as prefix. This is true for for example the
+#'   \code{xls.reader} and \code{xlsx.reader}.
+#'
+#' @export
+#'
+#' @seealso \code{\link{load.project}}, \code{\link{show.project}},
+#'    \code{\link{project.config}}
+#'
+#' @examples
+#' library('ProjectTemplate')
+#'
+#' \dontrun{list.data()}
+list.data <- function(override.config = NULL) {
+  config <- .load.config(override.config)
+  .list.data(config)
+}
+
+.list.data <- function(config) {
+  # Get list of variables in data/, always recursive to exclude cached
+  # variables from nested files
+  all.files <- list.files(path = 'data', recursive = TRUE, include.dirs = TRUE)
+  # Get list of variables according to configured recursive_loading, used as
+  # filtering variable later
+  data.files <- list.files(path = 'data', recursive = config$recursive_loading)
+
+  # Get variable name and reader from filenames
+  files.parsed <- .parse.extensions(all.files)
+  varnames <- files.parsed$varnames
+  readers <- files.parsed$readers
+
+  is_ignored <- grepl(.prepare.data.ignore.regex(config$data_ignore),
+                      all.files)
+  is_directory <- file.info(file.path('data', all.files))$isdir
+
+  is_cached <- .is.cached(varnames)
```

Comment:
initially all datasets from `data/` are listed, on line 81 all variables from the cache are loaded and matched to the corresponding set in `data/`. Variables cached during munging don't match a file, so end up in the second list.